resources/views/components/link-preview-card.blade.php

```diff
@@ -0,0 +1,50 @@
+<div
+    id="link-preview-card"
+    data-url="{{ $url }}"
+    class="mx-auto mt-2 min-w-full group/preview" data-navigate-ignore="true"
+>
+@if ($data->has('html'))
+    @if ($data->get('site_name') === 'X (formerly Twitter)')
+        @php($tweetId = explode('/', $url))
+        <div data-tweet-id="{{ end($tweetId) }}" class="w-full overflow-hidden">
+            {!! $data->get('html') !!}
```

Comment:
Is this safe?

Reply:
It's better if we prevent any bad actors from being able to add a og:html tag that will be parsed. So the only html keys will be from oEmbed calls from which we controller the source.

It's fixed by adding a list of allowed tags:
```php
$allowed = ['title', 'description', 'keywords', 'image', 'site_name', 'url', 'type'];
```

Then only adding the tag to the collection if its exists in the allowed array.
```php
  collect(['name', 'property'])
      ->map(fn ($name): string => $meta->getAttribute($name))
      ->filter(fn ($attribute): bool => in_array(explode(':', (string) $attribute)[0], $interested_in))
      ->each(function ($attribute) use ($data, $allowed, $meta): void {
          $key = explode(':', $attribute)[1];
          if (! $data->has($key) && in_array($key, $allowed, true)) {
              $data->put($key, $meta->getAttribute('content'));
          }
      });
```